overlord/snapstate/snapmgr.go

```diff
@@ -685,32 +684,18 @@ func (m *SnapManager) ensureUbuntuCoreTransition() error {
 	return nil
 }
 
-// GenerateCookies creates snap cookies for snaps that are missing them (may be the case for snaps installed
-// before the feature of running snapctl outside of hooks was introduced, leading to a warning
-// from snap-confine).
-// It is the caller's responsibility to lock state before calling this function.
-func (m *SnapManager) GenerateCookies(st *state.State) error {
-	var snapNames map[string]*json.RawMessage
-	if err := st.Get("snaps", &snapNames); err != nil && err != state.ErrNoState {
-		return err
-	}
-
-	var contexts map[string]string
-	if err := st.Get("snap-cookies", &contexts); err != nil {
-		if err != state.ErrNoState {
-			return fmt.Errorf("cannot get snap cookies: %v", err)
-		}
-		contexts = make(map[string]string)
-	}
+func (m *SnapManager) doSwitchSnap(t *state.Task, _ *tomb.Tomb) error {
```

Comment:
Hmm? What is going on here?

Reply:
This method has been moved to the separate cookies.go file.